backend/app/anti_recommendation_engine/anti_recommendation_engine.py

```diff
@@ -1,53 +1,72 @@
-from app.anti_recommenders.anti_recommender_generator import AntiRecommendationGenerator
-from app.models.record.record import Record
+from collections.abc import Generator
+
+from app.anti_recommenders.anti_recommender import AntiRecommender
+from app.anti_recommenders.open_ai.open_ai_normal_anti_recommender import (
+    OpenAiNormalAntiRecommender,
+)
+from app.models.anti_recommendation import AntiRecommendation
+from app.models.record import Record
+from app.models.settings import settings
 from app.readers.all_source_reader import AllSourceReader
 
 
-class AntiRecommendationEngine:
-    """
-    The main AntiRecommendationEngine for the NerdSwipe backend.
-    It retrieves AntiRecommendations from the AntiRecommendationGenerator and possesses methods
-    which return tuples of Records that have the same titles as the generated AntiRecommendations or previous AntiRecommendations.
+class AntiRecommendationEngine(AntiRecommender):
+    """AntiRecommendationEngine for the NerdSwipe backend.
+
+    Yields AntiRecommendations of a record_key,
+    Returns a tuple of Records that match the AntiRecommendations of a record_key,
+    Returns a tuple of Records that were the previous Records of AntiRecommendations.
     """
 
     def __init__(self) -> None:
-        self.__record_store: dict[str, Record] = self.load_records()
-        self.__anti_recommendation_generator: AntiRecommendationGenerator = (
-            AntiRecommendationGenerator()
+        self.__records_by_key: dict[str, Record] = self.load_records()
+        self.__anti_recommender: AntiRecommender | None = (
+            self.__select_anti_recommender()
         )
         self.__stack: list[list[Record]] = []
         self.__current_anti_recommendation_records: list[Record] = []
 
+    def __select_anti_recommender(self) -> AntiRecommender | None:
+        """Select and return an AntiRecommender based on values stored in settings."""
+
+        if (
+            settings.anti_recommender_type.lower() == "openai"
+            and settings.openai_api_key
+        ):
+            return OpenAiNormalAntiRecommender()
+
+        return None
+
     def load_records(self) -> dict[str, Record]:
```

Comment:
You don't need to make this public, or even have a separate method. You can just have these few lines of code in the __init__ method. For testing you could expose a simple `record_count` `@property` or similar.

Reply:
Yes, I removed `load_records()` and replaced it with a `dict` comprehension in the **init** method.